src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -56,18 +56,18 @@ TaskRevolutionParameters::TaskRevolutionParameters(PartDesignGui::ViewProvider*
     QMetaObject::connectSlotsByName(this);
     this->groupLayout()->addWidget(proxy);
 
-    //bind property mirrors
-    PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+    // bind property mirrors
+    auto pcFeat = dynamic_cast<PartDesign::ProfileBased *>(vp->getObject());
     if (pcFeat->isDerivedFrom(PartDesign::Revolution::getClassTypeId())) {
-        PartDesign::Revolution* rev = static_cast<PartDesign::Revolution*>(vp->getObject());
+        auto *rev = dynamic_cast<PartDesign::Revolution *>(vp->getObject());

```

Comment:
No need to replace the static_cast here because with the previous line this is already asserted.

Reply:
Yes, but the CI issues this warning nevertheless. What would be the disadvantage of using dynamic_cast?

Reply:
IMO we shouldn't blindly follow what the CI says. It only follows some static rules but doesn't know much about the context.  If the check `if (pcFeat->isDerivedFrom(PartDesign::Revolution::getClassTypeId()))` is true then it's already guaranteed that it's the right type.

The downside of using dynamic_cast here is that two expensive type checks are performed instead of only one.

Reply:
I understand but wanted to know how expensive this is. So my idea is maybe wrong: accepting a single additional type check for the sake of not being annoyed by warnings. (That catch attention of devs and in the end cost us unnecessary spare time.)

Reply:
> I understand but wanted to know how expensive this is

How expensive exactly I don't know. It depends on how the compiler implements it and the only way to check is write a test program and measure the time

Reply:
A simple test application shows that dynamic_cast is by a factor 4-5 slower than static_cast -- but it's still in the range of 10ns.

Since isDerivedFrom() is slower than dynamic_cast by  factor of 2.4 the most efficient way is to write:
```
if (auto *rev = dynamic_cast<PartDesign::Revolution *>(vp->getObject()) {
...
}
```
The second best way is to write
```
if (pcFeat->isDerivedFrom(PartDesign::Revolution::getClassTypeId())) {
    auto* rev = static_cast<PartDesign::Revolution*>(vp->getObject());
}
```
and the worst way is to write
```
if (pcFeat->isDerivedFrom(PartDesign::Revolution::getClassTypeId())) {
    auto* rev = dynamic_cast<PartDesign::Revolution*>(vp->getObject());
}
```

Reply:
Interesting. many thanks for this info.

I updated the code accordingly now.